setup.py

```diff
@@ -25,8 +27,36 @@
 # plain distutils when setuptools is not available.
 try:
     from setuptools import setup
+    from setuptools.command.install_scripts import install_scripts as orig
 except ImportError:
     from distutils.core import setup
+    from distutils.command.install_scripts import install_scripts as orig
+
+from distutils.file_util import copy_file
+from distutils.dir_util import mkpath
+from stat import ST_MODE
+
+class install_scripts(orig):
+    def run(self):
+        if sys.platform == 'win32':
+            super().run()
+            return
+
+        self.outfiles = []
+        if not self.dry_run:
+            mkpath(self.install_dir)
+
+        # We want the files to be installed without a suffix on Unix
+        for infile in self.get_inputs():
+            in_stripped = infile[:-3] if infile.endswith('.py') else infile
+            outfile = path.join(self.install_dir, in_stripped)
+            copy_file(infile, outfile, dry_run=self.dry_run)
+            self.outfiles.append(outfile)
+
+        if not self.dry_run:
+            for f in self.get_outputs():
+                mode = ((os.stat(f)[ST_MODE]) | 0o555) & 0o7777
```

Comment:
This was literally just a copy paste of the current behavior, I can look into `copy_file()` again and see if it gives us the option to preserve modes.